### DISCUSSION TITLE
Request to add new features to AdGuard Home :

### DISCUSSION THREAD

**Discussion by ghost** [AdguardTeam/AdGuardHome#3340] [Ideas]
Hello,

I take the liberty of contacting you, because I want to know if it is possible for you to add to AdGuard Home the possibility of :

- Configure "a proxy module" via the management web interface.
- Make "AdGuard Home encryption" automatic via the web interface by automatically generating an SSL certificate, a bit like with the AdGuard application for Windows.
- The possibility to "configure DNSCrypt" via the web interface of AdGuard Home.
- Finally, the possibility of "saving, restoring" the parameters of AdGuard Home directly via the web management interface this avoids wasting time going through the AdGuardHome.yaml folder.

I will be extremely happy to see these features added to the AdGuard Home management web interface because it is extremely complex to configure all of this via the AdGuard Home configuration file ;(

**Comment by ainar-g** [comment]
Hello.

I don't quite understand your first point.  What do you mean by “a proxy module“ and “make encryption automatic”?  The last point sounds a lot like #1147.  I'm not sure if the DNSCrypt configuration via the web interface will ever be used by a lot of people, especially since encrypted DNS moves towards DoT/DoH/DoQ.

@ameshkov, what do you think?

**Reply by ghost** [reply]
Hello Ainar-g, 

 The proxy module I mentioned and that of topic #1228.
 For automatic encryption, this is the encryption setting available on the AdGuard Home web management interface
 in fact, i want AdGuard Home to generate a certificate automatically so that we can use encryption support (HTTPS / TLS) for DNS and web administrator interface without having to configure it ourselves.
 Thanks, for topic #1147, I read it carefully;  I therefore await the implementation of this functionality.

 I'm sorry for the confusion in my first post, I misspoke.

**Comment by ameshkov** [comment]
Well, we do want to add all of these features in the future, it would take time though.
